src/shared/lib/file-validation.ts

```
@@ -0,0 +1,370 @@
+  if (nonPrintableRatio < 0.01 && controlRatio < 0.01 && highAsciiRatio < 0.1) {
+    return { isBinary: false, confidence: 0.9 };
+  }
+  
+  return { isBinary: false, confidence: 0.5 };
+}
+
+/**
+ * Validate MIME type against allowlist
+ */
+function validateMimeType(mimeType: string): boolean {
+  return ALLOWED_MIME_TYPES.includes(mimeType as any);
```

ESLint

[error] Unexpected any. Specify a different type.

(@typescript-eslint/no-explicit-any)